packages/lib/src/memoize_timeout.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Memoize a function with a timeout. Calling the function again after the timeout will erase the cache.
+ * @param fn Function to memoize
+ * @param time Time in ms to cache results
+ * @returns Memoized function
+ */
+// eslint-disable-next-line @typescript-eslint/ban-types
+export function memoizeTimeout(fn: Function, time: number): Function {
+	let cache = new Map();
+	let expiryTimers = new Map() as Map<string, NodeJS.Timeout>;
+
+	return (...args: any) => {
+		// Create hash.
+		const hash = JSON.stringify(args);
+
+		// Erase cache.
+		if (!expiryTimers.has(hash)) {
+			expiryTimers.set(hash, setTimeout(() => {
+				cache.delete(hash);
+				const interval = expiryTimers.get(hash);
+				clearInterval(interval);
+				expiryTimers.delete(hash);
+			}, time));
```

Comment:
When the timer here is created it'll keep the event loop alive when the program exits. So if you do `memoizeTimeout(foo, 300e3)` for example and the function is called before shutdown it'll hang the application for 5 minutes before exiting. Calling [`.unref()`](https://nodejs.org/api/timers.html#timeoutunref) on the timeout will prevent this from happening.

```suggestion
			expiryTimers.set(hash, setTimeout(() => {
				cache.delete(hash);
				const interval = expiryTimers.get(hash);
				clearInterval(interval);
				expiryTimers.delete(hash);
			}, time).unref());
```